src/lib/transition_router/transition_router.ml

```diff
@@ -139,6 +122,144 @@ module Make (Inputs : Inputs_intf) = struct
           ~transition_reader_ref ~transition_writer_ref ~frontier_w
           new_frontier )
 
+  let download_best_tip ~logger ~network ~verifier ~trust_system =
```

Comment:
Make sure when you see a better transition, write it into the pipe, `most_recent_valid_block_writer`. This would prevent us from getting weird stats showing up in coda status. Namely, it would not show us having a best tip length to be some value, like 17, and having the observed max block length to be 0.